zb-classifier.js

```diff
@@ -950,6 +959,401 @@ class ZigbeeClassifier {
+      coolTargetProperty.setMaximum(Math.max(max1, max2));
+    };
+
+    minCoolTargetProperty.updated = function() {
+      coolTargetProperty.setMinimum(this.value);
+    };
+
+    absMinCoolTargetProperty.updated = function() {
+      minCoolTargetProperty.setMinimum(this.value);
+    };
+
+    deadbandProperty.updated = function() {
+      heatTargetProperty.updateMinimum();
+      coolTargetProperty.updateMaximum();
+    };
+
+    this.addProperty(
+      node,                             // device
+      'hold',                           // name
+      {                                 // property description
+        '@type': 'BooleanProperty',
+        label: 'Hold',
+        type: 'boolean',
+      },
+      PROFILE_ID.ZHA,                   // profileId
+      hvacThermostatEndpoint,           // endpoint
+      CLUSTER_ID.HVACTHERMOSTAT,        // clusterId
+      'tempSetpointHold',               // attr
+      'setOnOffWriteValue',             // setAttrFromValue
+      'parseOnOffAttr',                 // parseValueFromAttr
+      CONFIG_REPORT_MODE
+    );
+
+    const uiCfgEndpoint =
+      node.findZhaEndpointWithInputClusterIdHex(
+        CLUSTER_ID.HVACUSERINTERFACECFG_HEX);
+
+    if (uiCfgEndpoint) {
+      this.addProperty(
+        node,                             // device
+        'units',                          // name
+        {                                 // property description
+          label: 'Units',
+          type: 'string',
+          enum: ['C', 'F'],
+        },
+        PROFILE_ID.ZHA,                   // profileId
+        uiCfgEndpoint,                    // endpoint
+        CLUSTER_ID.HVACUSERINTERFACECFG,  // clusterId
+        'tempDisplayMode',                // attr
+        'setWriteEnumValue',              // setAttrFromValue
+        'parseEnumAttr'                   // parseValueFromAttr
+      );
+    }
+
+    if (hvacFanControlEndpoint) {
+      const fanModeSeqProperty = this.addProperty(
+        node,                             // device
+        '_fanModeSeq',                    // name
+        {                                 // property description
+          label: 'Fan Sequence',
+          type: 'string',
+          readOnly: true,
+          enumArray: HVAC_FAN_SEQ,
+        },
+        PROFILE_ID.ZHA,                   // profileId
+        hvacFanControlEndpoint,           // endpoint
+        CLUSTER_ID.HVACFANCTRL,           // clusterId
+        'fanModeSequence',                // attr
+        '',                               // setAttrFromValue
+        'parseEnumAttr'                   // parseValueFromAttr
+      );
+      const fanModeProperty = this.addProperty(
+        node,                             // device
+        'fanMode',                        // name
+        {                                 // property description
+          label: 'Fan',
+          type: 'string',
+          enum: [],
+          enumArray: HVAC_FAN_MODE,
```

Comment:
s/enumArray/enum/

We don't support `enumArray`.

Reply:
Here, the enum is determined at runtime, enumArray is used internally to map the string to a value.